pyomo/contrib/pynumero/interfaces/tests/test_nlp.py

```diff
@@ -10,660 +10,609 @@
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+    # test inplace
+    jac_ineq.data = 0*jac_ineq.data
+    ret = anlp.evaluate_jacobian_ineq(out=jac_ineq)
+    self.assertTrue(ret is jac_ineq)
+    dense_jac_ineq = jac_ineq.todense()
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+    # change the value of the primals
+    # ToDo: not a great test since this problem is linear
+    anlp.set_primals(2.0*np.ones(9))
+    anlp.evaluate_jacobian_ineq(out=jac_ineq)
+    dense_jac_ineq = jac_ineq.todense()
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+
+    # hessian
+    hess = anlp.evaluate_hessian_lag()
+    dense_hess = hess.todense()
+    expected_hess = [ [2.0*i*j for j in range(1, 10)] for i in range(1,10) ]
+    expected_hess = np.asarray(expected_hess, dtype=np.float64)
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # test inplace
+    hess.data = np.zeros(len(hess.data))
+    ret = anlp.evaluate_hessian_lag(out=hess)
+    self.assertTrue(ret is hess)
+    dense_hess = hess.todense()
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # change the value of the primals
+    anlp.set_primals(2.0*np.ones(9))
+    anlp.evaluate_hessian_lag(out=hess)
+    dense_hess = hess.todense()
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # change the value of the obj factor
+    anlp.set_obj_factor(2.0)
+    hess = anlp.evaluate_hessian_lag()
+    dense_hess = hess.todense()
+    expected_hess = [ [4.0*i*j for j in range(1, 10)] for i in range(1,10) ]
+    expected_hess = np.asarray(expected_hess, dtype=np.float64)
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
 
-def create_rosenbrock_model(n_vars):
-
-    model = pe.ConcreteModel()
-    model.nx = n_vars
-    model.lb = -10
-    model.ub = 20
-    model.init = 2.0
-    model.index_vars = range(n_vars)
-    model.x = pe.Var(model.index_vars, initialize=model.init, bounds=(model.lb, model.ub))
-
-    def rule_constraint(m, i):
-        return m.x[i]**3-1.0 == 0
-    model.c = pe.Constraint(model.index_vars, rule=rule_constraint)
+@unittest.skipIf(os.name in ['nt', 'dos'], "Do not test on windows")
+class TestAslNLP(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.pm = create_pyomo_model1()
+        temporary_dir = tempfile.mkdtemp()
+        cls.filename = os.path.join(temporary_dir, "Pyomo_TestAslNLP")
+        cls.pm.write(cls.filename+'.nl', io_options={"symbolic_solver_labels": True})
 
-    model.obj = pe.Objective(expr=sum((model.x[i]-2)**2 for i in model.index_vars))
+    @classmethod
+    def tearDownClass(cls):
+        # TODO: remove the nl files
+        pass
+
+    def test_nlp_interface(self):
+        anlp = AslNLP(self.filename)
+        execute_extended_nlp_interface(self, anlp)
+        
+@unittest.skipIf(os.name in ['nt', 'dos'], "Do not test on windows")
```

Comment:
What part of PyNumero is not Windows compatible?

Reply:
Currently, I believe it is the pynumero_libraries that provide interfaces to ASL, etc. @eslickj may have a fix for this (he recently got PyNumero running on Windows). We would need to get these fixes into the conda recipe.

@santiagoropb, @eslickj: Is this interpretation correct?